rfcs/0000-omnilock/0000-omnilock.md

```diff
@@ -0,0 +1,558 @@
+        omni_identity:
+           identity: <flag: 0x0> <pubkey hash 2>
+           proofs: <SMT proofs for the above identity in AdminList Cell 1>
+        preimage: <EMPTY>
+      <...>
+```
+### Unlock via administrator's lock script hash (1)
+Note: the location of AdminList Cell 1 is in cell deps
+
+```
+CellDeps:
+    <vec> Omnilock Script Cell
+    <vec> AdminList Cell 1
+Inputs:
+    <vec> Cell
+        Data: <...>
+        Type: <...>
+        Lock:
+            code_hash: Omnilock
+            args: <flag: 0> <pubkey hash 1> <Omnilock flags: 1> <AdminList Cell 1's type ID>
+    <vec> Cell
+        Data: <...>
+        Type: <...>
+        Lock: blake160 for this lock script must be 0x12...34
+    <...>
+Outputs:
+    <vec> Any cell
+Witnesses:
+    WitnessArgs structure:
+      Lock:
+        signature: <EMPTY>
+        omni_identity:
+           identity: <flag: 0xFC> <lock hash: 0x12...34>
+           proofs: <SMT proofs for the above identity in AdminList Cell 1>
+        preimage: <EMPTY>
+      <...>
+```
+
+### Unlock via administrator's lock script hash (2)
+Note: the location of AdminList Cell 1 is in input cell
+
+```
+CellDeps:
+    <vec> Omnilock Script Cell
+
+Inputs:
+    <vec> Cell
+        Data: <...>
+        Type: <...>
+        Lock:
+            code_hash: Omnilock
+            args: <flag: 0> <pubkey hash 1> <Omnilock flags: 1> <AdminList Cell 1's type ID>
+    <vec> Cell
+        Data: <...>
+        Type: <...>
+        Lock: blake160 for this lock script must be 0x12...34
+    <vec> AdminList Cell 1
+        Data: <RCData, union of RCCellVec and RCRule>
+        Type: <its hash is same to AdminList Cell 1's type ID>
+        Lock: <...>
+    <...>
+Outputs:
+    <vec> Any cell
+Witnesses:
+    WitnessArgs structure:
+      Lock:
+        signature: <EMPTY>
+        omni_identity:
+           identity: <flag: 0xFC> <lock hash: 0x12...34>
+           proofs: <SMT proofs for the above identity in AdminList Cell 1>
+        preimage: <EMPTY>
+      <...>
+```
+
+### Unlock via anyone-can-pay
+
+```
+CellDeps:
+    <vec> Omnilock Script Cell
+Inputs:
+    <vec> Cell
+        Data: <...>
+        Type: <...>
+        Lock:
+            code_hash: Omnilock
+            args: <flag: 0x0> <pubkey hash 1> <Omnilock flags: 2> <2 bytes minimun ckb/udt in ACP>
+    <...>
+    follow anyone-can-pay rules
+    <...>
+Outputs:
+    <vec> Any cell
+Witnesses:
+    WitnessArgs structure:
+      Lock:
+        signature: <EMPTY>
+        omni_identity: <EMPTY>
+        preimage: <EMPTY>
+      <...>
+```
+
+### Unlock via dynamic linking
+
+```
+CellDeps:
+    <vec> Omnilock Script Cell
+Inputs:
+    <vec> Cell
+        Data: <...>
+        Type: <...>
+        Lock:
+            code_hash: Omnilock
+            args: <flag: 0xFE> <preimage hash> <Omnilock flags: 0>
+    <...>
+Outputs:
+    <vec> Any cell
+Witnesses:
+    WitnessArgs structure:
+      Lock:
+        signature: <valid secp256k1 signature for pubkey hash 1>
```

Comment:
```suggestion
        signature: <valid delegate signature for pubkey hash 1>
```

I found that in the current omnilock implementation, there is [no restriction](https://github.com/nervosnetwork/ckb-production-scripts/blob/master/c/ckb_identity.h#L343-L344) that the cryptographic algorithm must be secp256k1 elliptic curve, or even [other cryptographic algorithms](https://github.com/nervosnetwork/ckb-production-scripts/blob/master/tests/omni_lock_rust/tests/misc.rs#L633-L635)

Reply:
This is an example so that it's described as detailed as possible.

Reply:
as a newcomer, I may mistakenly think that Omnilock can only support secp256k1, since secp256k1 is very common in CKB, maybe we can replace this example with RSA to explain the meaning of **delegate signature**?

Reply:
As an example, it should be explained as detailed as possible. This is not only applied to this RFC but also other RFCs.
To understand RFC fully, newcomers need to read chapters above.